project/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,34 +72,44 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
    * formats the data as {@code Election} objects. Correlates one {@code Election} with one or more
    * {@code Position}.
    */
-  private List<Election> extractElectionInformation(String address, boolean listAll) {
+  private List<Election> extractElectionInformation(String address, boolean listAllElections) {
     List<Election> elections = new ArrayList<>();
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query electionQuery = new Query("Election");
     PreparedQuery electionQueryResult = datastore.prepare(electionQuery);
     List<Entity> electionsData = electionQueryResult.asList(FetchOptions.Builder.withDefaults());
     for (Entity election : electionsData) {
-      if (!isRelevantElection(election, address, listAll)) {
+      if (!isRelevantElection(election, address, listAllElections)) {
+        continue;
+      }
+      List<String> candidatePositionsData = 
+          (List<String>) election.getProperty("candidatePositions");
+      List<String> candidateIdsData = 
+          (List<String>) election.getProperty("candidateIds");
+      List<Boolean> candidateIncumbencyData = 
+          (List<Boolean>) election.getProperty("candidateIncumbency");
+      if (candidatePositionsData == null || candidateIdsData == null 
+          || candidateIncumbencyData == null) {
         continue;
       }
       List<Position> positions =
-          extractPositionInformation((List<String>) election.getProperty("candidatePositions"),
-          (List<String>) election.getProperty("candidateIds"),
-          (List<Boolean>) election.getProperty("candidateIncumbency"));
+          extractPositionInformation(candidatePositionsData, candidateIdsData,
+                                    candidateIncumbencyData);
       elections.add(new Election(election.getKey().getName(),
-          (Date) election.getProperty("date"), positions));
+                                (Date) election.getProperty("date"), positions));
```

Comment:
extra small nit: align the parameters, meaning that `e` should be aligned with `(`